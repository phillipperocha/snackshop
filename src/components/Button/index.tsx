import React from 'react'

const Button: React.FC = () => <button type="button">Snack Shop</button>

export default Button
