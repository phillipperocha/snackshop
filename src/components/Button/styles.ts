import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.weight.black};
    color: ${theme.colors.white};
    cursor: pointer;
    border: none;
    border-radius: ${theme.border.radius};
    text-decoration: none;
    align-items: center;
    justify-content: center;
    height: 6.1rem;
    width: 100%;
    text-transform: uppercase;
    padding: 0.8rem;
  `}
`
