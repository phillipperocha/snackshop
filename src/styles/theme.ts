export default {
  border: {
    radius: '5px',
  },
  font: {
    family: "'Roboto', sans-serif",
    weight: {
      normal: 400,
      bold: 700,
      black: 900,
    },
    sizes: {
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '3.0rem',
    },
  },
  colors: {
    primary: '#FF0000',
    mainBg: '#FFFFFF',
    white: '#FFFFFF',
    black: '#333333',
    lightGray: '#DDDDDD',
    gray: '#AEAEAE',
    darkGray: '#707070',
  },
  spacings: {
    xxsmall: '1.0rem',
    xsmall: '2.0rem',
    small: '2.4rem',
    medium: '3.5rem',
  },
}
