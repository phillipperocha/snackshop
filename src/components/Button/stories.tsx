import { Story, Meta } from '@storybook/react/types-6-0'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    children: {
      type: 'string',
    },
  },
} as Meta

export const Default: Story = (args) => (
  <div style={{ width: '42.5rem' }}>
    <Button {...args} />
  </div>
)

Default.args = {
  children: 'Carrinho',
}
