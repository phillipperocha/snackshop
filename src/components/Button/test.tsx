import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import theme from 'styles/theme'

import Button from '.'

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = renderWithTheme(<Button>CARRINHO</Button>)

    expect(screen.getByRole('button', { name: /carrinho/i })).toHaveStyle({
      height: '6.1rem',
      color: theme.colors.white,
      background: theme.colors.primary,
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
