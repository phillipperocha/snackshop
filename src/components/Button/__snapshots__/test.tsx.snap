// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
.c0 {
  background: #FF0000;
  font-family: 'Roboto',sans-serif;
  font-size: 2.0rem;
  font-weight: 900;
  color: #FFFFFF;
  cursor: pointer;
  border: none;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 6.1rem;
  width: 100%;
  text-transform: uppercase;
  padding: 0.8rem;
}

<button
  class="c0"
>
  CARRINHO
</button>
`;
